lib/geometry.rb

```diff
@@ -20,29 +20,76 @@ def distance_point_to_point2d(a, b)
       Math.sqrt((b.x - a.x)**2 + (b.y - a.y)**2)
     end
 
-    # Class represents a line as equation y = k*x +b
-    # k - slope
-    # b - free_term
+    ##
+    # Class represents a line as equation Ax + By + C = 0
     # in two-dimensional space
     class Line2dCanon
-      attr_reader :slope
-      attr_reader :free_term
+      attr_reader :x_coefficient
+      attr_reader :y_coefficient
+      attr_reader :free_coefficient
 
-      def initialize(p1, p2)
-        if (p1.x == p2.x) && (p1.y == p2.y)
-          raise ArgumentError, "You need 2 diffrent points"
+      def initialize(point1, point2)
```

Comment:
This method totally needs RDoc-style comment to make sure that ```point1``` and ```point2``` are supposed to have methods ```x``` and ```y```.